packages/pwa-kit-dev/src/utils/script-utils.ts

```diff
@@ -112,6 +113,53 @@ export const walkDir = async (
     return fileSet
 }
 
+export const getAllProjectDependencyVersions = (): {[x: string]: any} => {
+    /**
+     * Returns the versions of all packages and their dependencies within the project.
+     *
+     * @returns An object representing the dependency tree
+     */
+    return JSON.parse(execSync(`npm ls --all --json`, {encoding: 'utf-8'}))
+}
+
+export const getLowestPackageVersion = (
+    packageName: string,
+    dependencyTree: {[x: string]: any}
+): string => {
+    /**
+     * Returns the lowest version of a package installed.
+     *
+     * @param packageName - The name of the package to get the lowest version for
+     * @param dependencyTree - The dependency tree including all package versions
+     * @returns The lowest version of the given package that is installed
+     */
+    let lowestVersion: string | null = null
+
+    function search(dependencyTree: {[x: string]: any}) {
+        for (const key in dependencyTree) {
+            if (key === packageName && dependencyTree[key].version) {
+                const version = dependencyTree[key].version
+                if (!lowestVersion || version < lowestVersion) {
+                    lowestVersion = version
+                }
+            }
+
+            if (typeof dependencyTree[key] === 'object') {
+                search(dependencyTree[key])
+            }
+        }
+    }
+
+    search(dependencyTree)
+
+    if (lowestVersion) {
+        return lowestVersion
+    } else {
+        console.warn(`Unable to find package version for ${packageName}`)
+        return 'unknown'
+    }
```

Comment:
```suggestion
   search(dependencyTree)
    return lowestVersion ?? 'unknown'
```
Mostly for the console message - it will be displayed to customers when they do the push command. They don't care that we can't find it, and it may cause confusion.

Reply:
Wow didn't know you could do `??` in javascript😮